### ISSUE TITLE
[wasm] Sanity check generated nuget packages

### ISSUE THREAD

**Issue by radical** [dotnet/runtime#50198]
We recently had a regression where `mono-aot-cross.exe` got dropped from the cross compiler package on windows.
We should have some basic checks like validating the list of included files against an expected list, to guard against this in the future.

AFAICS, we don't have anything for that right now.

/cc @akoeplinger @steveisok @lewing

**Comment by lewing** [comment]
@steveisok it happened again 🤦  can we get some validation setup this time?

**Comment by lewing** [comment]
```
/usr/local/share/dotnet/packs/Microsoft.NET.Runtime.WebAssembly.Sdk/6.0.0-preview.5.21262.1/Sdk/WasmApp.targets(358,5): error : Could not find AOT cross compiler at $(MonoAotCrossCompilerPath)=/usr/local/share/dotnet/packs/Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.browser-wasm/6.0.0-preview.5.21262.1/Sdk/../tools/mono-aot-cross [/Users/lewing/Source/bsize/bsize.csproj]```

**Comment by radical** [comment]
Would the tests under https://github.com/dotnet/runtime/tree/main/src/libraries/pkg/test be suitable for this? IIUC, we would add a targets file under `src/libraries/pkg/test/packageSettings/$packageId/`, which could then do the check.

**Comment by steveisok** [comment]
@directhex does it make sense to add pkg tests here?

**Comment by eerhardt** [comment]
I would think we would want tests that tested the end-to-end experience like a customer would use the product. That way, we can ensure all the parts work together correctly.

Given that reasoning, I think the tests would probably live in `dotnet/sdk` or `dotnet/installer`. We have similar tests in those repos that test the end-to-end, for example for Blazor WASM apps.

https://github.com/dotnet/sdk/blob/23e7fd3c08395b368f7db8bee30c20c558b29bd4/src/Tests/Microsoft.NET.Sdk.BlazorWebAssembly.Tests/WasmPublishTest.cs#L25

**Comment by steveisok** [comment]
Agreed.  I would also like to test coming out of runtime as certain small build tweaks can end up causing sizable regressions.

**Comment by eerhardt** [comment]
![image](https://user-images.githubusercontent.com/8291187/118337174-d2f2df80-b4d8-11eb-96ec-f51aab229f73.png)

**Comment by lewing** [comment]
> Given that reasoning, I think the tests would probably live in `dotnet/sdk` or `dotnet/installer`. We have similar tests in those repos that test the end-to-end, for example for Blazor WASM apps.

The Manifest is consumed by directly by installer as part of the baseline manifests so that it probably a good place for an e2e check

**Comment by directhex** [comment]
This is resolved by https://github.com/dotnet/runtime/pull/52784

**Comment by eerhardt** [comment]
> This is resolved by #52784

I don't see tests being added with that change. What are we doing to ensure this doesn't regress again?
